src/main/java/org/apache/sysml/api/python/SystemML.py

```diff
@@ -37,11 +37,329 @@
 from pyspark.ml import Estimator, Model
 from scipy.sparse import spmatrix
 from scipy.sparse import coo_matrix
+from contextlib import contextmanager
+
+def simpleAddPyDMLStmt(fnName, args=None, obj=None):
+    m = Matrix(0)
+    addPyDMLStmt(fnName, lhs=m, args=args, obj=obj)
+    return m
+
+def addPyDMLStmt(fnName, lhs=None, args=None, obj=None):
+    pydmlStmt = [getTabs()]
+    if lhs is not None:
+        pydmlStmt = pydmlStmt + [str(lhs.getID()), ' = ']
+    if obj is None:
+        pydmlStmt = pydmlStmt + [ fnName, '(']
+    else:
+        pydmlStmt = pydmlStmt + [  obj, '.', fnName, '(']
+    if args is not None:
+        pydmlStmt = pydmlStmt + args
+    MLContext.globalMLContext.pydml.append(''.join(pydmlStmt + [')\n']))
+ 
+def addPyDMLString(pydmlStmt):
+    MLContext.globalMLContext.pydml.append(''.join(pydmlStmt))
+   
+# Constructor         
+def full(init, rows=None, cols=None):
+    if rows is None or cols is None:
+        raise Exception('rows and cols are mandatory arguments')
+    scopeCheck('full')
+    return simpleAddPyDMLStmt('full', args=[str(init), ', rows=', str(rows), ', cols=', str(cols)])
 
 SUPPORTED_TYPES = (np.ndarray, pd.DataFrame, spmatrix)
 
-class MLContext(object):
+def matrix(numPyArr):
+    scopeCheck('matrix')
+    if isinstance(numPyArr, np.ndarray):
+        colNames = []
+        numCols = len(numPyArr[0,:])
+        for i in range(0, numCols):
+            colNames = colNames + [ str('C' + str(i))]
+        pdf = pd.DataFrame(numPyArr, columns=colNames)
+        df = MLContext.globalMLContext.sqlCtx.createDataFrame(pdf, colNames)
+    elif isinstance(numPyArr, pd.core.frame.DataFrame):
```

Comment:
Combine this with the NumPy if clause above, and just simply convert the `numPyArr` to a Pandas DataFrame in either case, regardless if it is a Numpy array, or Pandas DataFrame.  `pd.DataFrame` won't make a copy of the data for Numpy or Pandas DataFrame inputs anyways, so no harm done.  Also, we should add Python lists as well, and also convert those.  So, basically, this eliminates this additional clause.